src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs

```diff
@@ -61,6 +67,36 @@ await _documentClient.CreateDatabaseIfNotExistsAsync(new Database {Id = _options
                 new RequestOptions {OfferThroughput = _options.Value.OfferThroughput}).ConfigureAwait(false);
         }
 
+        private async Task CreateLeaseRequestsCollection()
+        {
+            var collectionDefinition = new DocumentCollection
+            {
+                Id = _options.Value.RequestsCollection
+            };
+
+            var leaseConstraint = new UniqueKey
+            {
+                Paths = new Collection<string> { "/instanceId", "/leaseType" }
+            };
+
+            var sortingIndex = new Collection<CompositePath>
+            {
+                new CompositePath { Path = "/priority", Order = CompositePathSortOrder.Ascending },
+                new CompositePath { Path = "/_ts", Order = CompositePathSortOrder.Ascending }
+            };
+
+            collectionDefinition.UniqueKeyPolicy.UniqueKeys.Add(leaseConstraint);
+            collectionDefinition.IndexingPolicy.CompositeIndexes.Add(sortingIndex);
+
+            collectionDefinition.PartitionKey.Paths.Add("/leaseType");
+            collectionDefinition.DefaultTimeToLive = 30;
```

Comment:
Can we ensure that TTL is set per document instance, please? Not sure if we want to do it on collection level?

Reply:
@faprovitolo Can you add a TTL property to the LeaseRequest document in your task?

Reply:
sure @nikhilgirraj but it should considered that, according to Microsoft docs:
"Time to Live on an item (set using ttl):

- This Property is applicable only if DefaultTimeToLive is present and it is not set to null for the parent container.

- If present, it overrides the DefaultTimeToLive value of the parent container."

I can possibly add TTL to each item setting it to 30 sec while setting defaulttimtolive to -1?

Reply:
The lease request delay is driven by config, isn't it? So it cannot be set here but if a value has to be set, it's fine if we default it to something - I assumed we can only set it when persisting the document.